File: frontend/src/services/doctorService.js
import apiClient from './apiClient';

const doctorService = {
  async getAll(params = {}) {
    const { data } = await apiClient.get('/doctors', { params });
    return data.data; // PageResponse
  },
  async getById(id) {
    const { data } = await apiClient.get(`/doctors/${id}`);
    return data.data;
  },
  async getByUserId(userId) {
    const { data } = await apiClient.get(`/doctors/by-user/${userId}`);
    return data.data;
  },
  async getByDepartment(departmentId) {
    const { data } = await apiClient.get(`/doctors/department/${departmentId}`);
    return data.data;
  },
  async search(keyword) {
    const { data } = await apiClient.get('/doctors/search', { params: { keyword } });
    return data.data;
  },
  async create(payload) {
    const { data } = await apiClient.post('/doctors', payload);
    return data.data;
  },
  async update(id, payload) {
    const { data } = await apiClient.put(`/doctors/${id}`, payload);
    return data.data;
  },
  async setAvailability(id, available) {
    const { data } = await apiClient.patch(`/doctors/${id}/availability`, null, { params: { available } });
    return data.data;
  },
  async remove(id) {
    await apiClient.delete(`/doctors/${id}`);
  },
};

export default doctorService;

File: backend/src/main/java/com/hospital/hms/service/DoctorService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.DoctorRequest;
import com.hospital.hms.dto.response.DoctorResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface DoctorService {

    DoctorResponse create(DoctorRequest request);

    DoctorResponse update(Long id, DoctorRequest request);

    DoctorResponse getById(Long id);

    DoctorResponse getByUserId(Long userId);

    Page<DoctorResponse> getAll(Pageable pageable);

    List<DoctorResponse> getByDepartment(Long departmentId);

    List<DoctorResponse> search(String keyword);

    void delete(Long id);

    void setAvailability(Long id, boolean available);
}
